src/classes/Transaction.ts

```diff
@@ -0,0 +1,92 @@
+import { AbstractProvider, JsonRpcSigner, PreparedTransactionRequest } from 'ethers'
+
+export class Transaction {
+    /**
+     * Constructs a new Transaction instance.
+     *
+     * @param transaction - PreparedTransactionRequest object containing transaction details.
+     * @param signer - Signer object to sign and send the transaction.
+     */
+    public constructor(
+        public readonly transaction: PreparedTransactionRequest,
```

Comment:
I'd suggest to rename this to `preparedTransactionRequest`, to match the interface that this parameter is representing.

Or at least use a different name than it's being used to declare the class. Semantically speaking is strange that the `Transaction` receives a `transaction`